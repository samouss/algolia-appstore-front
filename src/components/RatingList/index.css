.RatingList {
  min-height: 137px;
}
