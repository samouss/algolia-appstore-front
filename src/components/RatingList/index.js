import React from 'react';
import PropTypes from 'prop-types';
import FilterList from 'components/FilterList';
import FilterItem from 'components/FilterItem';
import Stars from 'components/Stars';
import Counter from 'components/Counter';
import style from './index.css';

const RatingList = ({ facetValues, onChange }) => (
  <FilterList
    customClassName={style.RatingList}
  >
    {facetValues.map(({ name, count, isRefined }) => (
      <FilterItem
        key={name}
        isRefined={isRefined}
        onChange={() => onChange(name, isRefined)}
      >
        <Stars
          value={parseInt(name, 10)}
          on={5}
        />

        <Counter
          value={count}
        />
      </FilterItem>
    ))}
  </FilterList>
);

RatingList.propTypes = {
  facetValues: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    isRefined: PropTypes.bool.isRequired,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default RatingList;
