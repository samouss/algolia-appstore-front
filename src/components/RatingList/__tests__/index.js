import React from 'react';
import { shallow } from 'enzyme';
import RatingList from '../index';

describe('<RatingList />', () => {
  const defaultProps = {
    facetValues: [
      {
        name: '4',
        count: 1234,
        isRefined: true,
      },
      {
        name: '2',
        count: 5678,
        isRefined: false,
      },
    ],
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <RatingList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <RatingList
        {...props}
      />,
    );

    component
      .find('FilterItem')
      .first()
      .simulate('change');

    expect(props.onChange).toHaveBeenCalledWith(
      '4',
      true,
    );
  });
});
