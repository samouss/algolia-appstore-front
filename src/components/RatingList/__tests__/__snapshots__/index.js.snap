// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingList /> expect to render 1`] = `
<FilterList
  customClassName="RatingList"
>
  <FilterItem
    customClassName=""
    isRefined={true}
    onChange={[Function]}
  >
    <Stars
      color=""
      on={5}
      value={4}
    />
    <Counter
      customClassName=""
      value={1234}
    />
  </FilterItem>
  <FilterItem
    customClassName=""
    isRefined={false}
    onChange={[Function]}
  >
    <Stars
      color=""
      on={5}
      value={2}
    />
    <Counter
      customClassName=""
      value={5678}
    />
  </FilterItem>
</FilterList>
`;
